### PR TITLE
APC code deep cleaning

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Split the apc code into several files for easier reading/maintaining
Cleanup the code by using early returns and changing some var names to the appropriate ones

## Why It's Good For The Game
code maintainability is good

### CODE REVIEW

File: code/__DEFINES/apc_defines.dm [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+/// Bit shift for the lighting channel status of the APC.
+#define UPOVERLAY_LIGHTING_SHIFT (6)
+/// Bit shift for the environment channel status of the APC.
+#define UPOVERLAY_ENVIRON_SHIFT (8)
+
+#define ACP_AI_NO_MALF 0
+#define ACP_AI_NO_HACK 1
+#define ACP_AI_HACK_NO_SHUNT 2
+#define ACP_AI_HACK_SHUNT_HERE 3
+#define ACP_AI_HACK_SHUNT_ANOTHER 4
```
Comment (TemporalOroboros) [inline code/__DEFINES/apc_defines.dm:109]:
```

#define APC_AI_NO_MALF 0
#define APC_AI_NO_HACK 1
#define APC_AI_HACK_NO_SHUNT 2
#define APC_AI_HACK_SHUNT_HERE 3
#define APC_AI_HACK_SHUNT_ANOTHER 4
```

File: code/modules/power/apc/apc_malf.dm [hunk 2]
```diff
@@ -0,0 +1,131 @@
+/obj/machinery/power/apc/proc/get_malf_status(mob/living/silicon/ai/malf)
+	if(!istype(malf) || !malf.malf_picker)
+		return ACP_AI_NO_MALF
+	if(malfai != (malf.parent || malf))
+		return ACP_AI_NO_HACK
+	if(occupier == malf)
+		return ACP_AI_HACK_SHUNT_HERE
+	if(istype(malf.loc, /obj/machinery/power/apc))
+		return ACP_AI_HACK_SHUNT_ANOTHER
+	return ACP_AI_HACK_NO_SHUNT
```
Comment (TemporalOroboros) [inline code/modules/power/apc/apc_malf.dm:10]:
```
/obj/machinery/power/apc/proc/get_malf_status(mob/living/silicon/ai/malf)
	if(!istype(malf) || !malf.malf_picker)
		return APC_AI_NO_MALF
	if(malfai != (malf.parent || malf))
		return APC_AI_NO_HACK
	if(occupier == malf)
		return APC_AI_HACK_SHUNT_HERE
	if(istype(malf.loc, /obj/machinery/power/apc))
		return APC_AI_HACK_SHUNT_ANOTHER
	return APC_AI_HACK_NO_SHUNT
```

File: code/__DEFINES/apc_defines.dm (hunk 1 above)
Comment (Ghilker) [reply code/__DEFINES/apc_defines.dm:109]: thanks for noticing lol

File: code/modules/power/apc/apc_main.dm [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+		return
+	if(cell && cell.charge >= 20)
+		cell.use(20)
+		INVOKE_ASYNC(src, .proc/break_lights)
+
+/obj/machinery/power/apc/proc/break_lights()
+	for(var/obj/machinery/light/breaked_light as anything in area)
+		breaked_light.on = TRUE
+		breaked_light.break_light_tube()
+		stoplag()
```
Comment (SabreML) [inline code/modules/power/apc/apc_main.dm:552]: Just a heads up, the `as anything in area` here is causing runtimes. (On a downstream technically, but the code is untouched.)
image

File: code/modules/power/apc/apc_power_proc.dm [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	for(var/mob/living/silicon/ai as anything in GLOB.ai_list)
+		if(get_area(ai) == area)
+			return
+
+	failure_timer = max(failure_timer, round(duration))
+
+/obj/machinery/power/apc/proc/set_nightshift(on)
+	set waitfor = FALSE
+	nightshift_lights = on
+	for(var/obj/machinery/light/night_light as anything in area)
```
Comment (SabreML) [inline code/modules/power/apc/apc_power_proc.dm:126]: Might want to look into changing these too.

**Pull request merged**